src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/String/KeyAnalyzer.cs

```diff
@@ -238,5 +181,49 @@ private static double GetUniquenessFactor(HashSet<string> set, ReadOnlySpan<stri
 
             return set.Count / (double)uniqueStrings.Length;
         }
+
+        internal record struct AnalysisResults(
```

Comment:
Please avoid using records.  They come with a whole bunch of extra gook we don't need.